pyo3-macros-backend/src/introspection.rs

```diff
@@ -184,7 +219,7 @@ fn argument_introspection_data<'a>(
 
 enum IntrospectionNode<'a> {
     String(Cow<'a, str>),
-    IntrospectionId(Option<&'a Ident>),
+    IntrospectionId(Option<Type>),
```

Comment:
Given the presence of lifetimes already here, should we make this `Option<Cow<'a, Type>>`?

Reply:
Indeed. Done.